adr/0004-application-architecture.md

```diff
@@ -0,0 +1,30 @@
+# 4. Application architecture
+
+Date: 2019-09-25
+
+## Status
+
+Accepted
+
+## Context
+
+We are starting to build a beta of the 'Apply for postgraduate teacher training' service. This service will have 3 distinct user groups, who will have their own interface in the service.
+
+- Candidates are people who would like to become a teacher. The service
+- Provider users work for an organisation that provides teacher training, like a school or university
+- Vendors make software that universities use for administration
+
+For the architecture of the system, we've discussed 2 options: a microservice architecture and a monolith.
+
+In the microservice architecture, we might split up the service into 3 or more applications - for example an API, a frontend for candidates and a frontend for the provider users. In the monolithic architecture, all components live together in a single application.
+
+## Decision
+
+We've chosen to go for a monolith. After discussing the options, we hypothesise that the microservice architecture for this project doesn't provide many of the benefits often associated with microservice approach (for [example](https://rubygarage.org/blog/advantages-of-microservices)). For example, the components are very tightly coupled and will almost always change together rather than independently. Using a monolith approach will make the project easier to run for developers and easier to deploy.
```

Comment:
```suggestion
We've chosen to go for a monolith. After discussing the options, we hypothesise that the microservice architecture for this project doesn't provide many of the [benefits often associated with microservice approach](https://rubygarage.org/blog/advantages-of-microservices).

For example, the components are very tightly coupled and will almost always change together rather than independently. Using a monolith approach will make the project easier to run for developers and easier to deploy.
```